test/unit/template_importer_test.rb

```diff
@@ -0,0 +1,96 @@
+require 'test_plugin_helper'
+
+module ForemanTemplates
+  class TemplateImporterTest < ActiveSupport::TestCase
+    def get_template(name)
+      File.expand_path(File.join('..', '..', 'templates', name), __FILE__)
+    end
+
+    def importer(opts = {})
+      # This uses a checkout of the plugin as a source of templates Caveat:
+      # changes to /test/templates will need to be committed (locally) for
+      # tests to work
+
+      path = File.expand_path(File.join('..','..','..'),__FILE__)
+      ForemanTemplates::TemplateImporter.new({
+        :repo      => path,
+        :branch    => Git.open(path).current_branch,
+        :prefix    => 'FooBar ',
+        :dirname   => '/test/templates',
+        :verbose   => 'false',
+        :associate => 'new'
+      }.merge(opts))
+    end
+
+    setup do
+      @importer = importer
+    end
+
+    context 'get_default_branch' do
+      setup do
+        @repo = Struct.new(:branches).new([
+          Struct.new(:name).new('0.1-stable'),
+          Struct.new(:name).new('develop')
+        ])
+      end
+
+      test 'when on develop, returns develop' do
+        SETTINGS[:version].stubs(:tag).returns('develop')
+        assert_equal 'develop', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch exists, return it' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.1')
+        assert_equal '0.1-stable', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch does not exist, use default branch' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.2')
+        refute @importer.get_default_branch(@repo)
+      end
+    end
+
+    test 'metadata method extracts correct metadata' do
+      text = File.read(get_template('metadata1.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'provision', hash['kind']
+      assert_equal 'Test Data', hash['name']
+      assert_equal 5, hash['oses'].size
+
+      text = File.read(get_template('metadata2.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'ProvisioningTemplate', hash['model']
+    end
+
+    test 'import! loads templates into db' do
```

Comment:
Hi Greg,

  Made the changes, not pushed yet as I can't run the tests and I'm don't
know what to do about this error:

bundle install
bundle exec rake

~/.gem/ruby/gems/bundler-1.11.2/lib/bundler/rubygems_integration.rb:304:in
`block in replace_gem': rake is not part of the bundle. Add it to Gemfile.
(Gem::LoadError)
        from /home/vagrant/bin/rake:22:in `<main>'


Dev box:
OEL 7.2
Ruby ruby 2.0.0p598 (2014-11-13) [x86_64-linux]


Any pointers appreciated,

On 31 August 2016 at 21:32, Greg Sutcliffe <notifications@github.com> wrote:

> In test/unit/template_importer_test.rb
> <https://github.com/theforeman/foreman_templates/pull/29#discussion_r77067383>
> :
>
> > +      end
> > +    end
> > +
> > +    test 'metadata method extracts correct metadata' do
> > +      text = File.read(get_template('metadata1.erb'))
> > +      hash = @importer.parse_metadata(text)
> > +      assert_equal 'provision', hash['kind']
> > +      assert_equal 'Test Data', hash['name']
> > +      assert_equal 5, hash['oses'].size
> > +
> > +      text = File.read(get_template('metadata2.erb'))
> > +      hash = @importer.parse_metadata(text)
> > +      assert_equal 'ProvisioningTemplate', hash['model']
> > +    end
> > +
> > +    test 'import! loads templates into db' do
>
> No worries. I have one more test to add to #27
> <https://github.com/theforeman/foreman_templates/pull/27> and then I can
> merge it, which will make your fork simple again ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman_templates/pull/29/files/896c6f18905e9e84edcd9ca2eed82dedb70761ad#r77067383>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABx8slFdkmYMURg1p5VSbY2Ie3DaC9EPks5qleTkgaJpZM4JxgHW>
> .
>